katharsis-client/src/main/java/io/katharsis/client/internal/ClientResourceUpsert.java

```diff
@@ -0,0 +1,176 @@
+		}
+	}
+
+	/**
+	 * Get relations from includes section or create a remote proxy
+	 */
+	@Override
+	protected Object fetchRelatedObject(RegistryEntry entry, Serializable relationId, RepositoryMethodParameterProvider parameterProvider, QueryAdapter queryAdapter) {
+
+		String uid = getUID(entry, relationId);
+		Object relatedResource = resourceMap.get(uid);
+		if (relatedResource != null) {
+			return relatedResource;
+		} else {
+			ResourceInformation resourceInformation = entry.getResourceInformation();
+			Class<?> resourceClass = resourceInformation.getResourceClass();
+			String url = null;
+			return proxyFactory.createResourceProxy(resourceClass, relationId, url);
+		}
+	}
+
+	public List<Object> allocateResources(List<Resource> resources) {
+		List<Object> objects = new ArrayList<>();
+		for (Resource resource : resources) {
+
+			RegistryEntry<?> registryEntry = resourceRegistry.getEntry(resource.getType());
+			ResourceInformation resourceInformation = registryEntry.getResourceInformation();
+
+			Object object = newResource(resourceInformation, resource);
+			setId(resource, object, resourceInformation);
+			setAttributes(resource, object, resourceInformation);
+			setLinks(resource, object, resourceInformation);
+			setMeta(resource, object, resourceInformation);
+
+			objects.add(object);
+
+			String uid = getUID(resource);
+			resourceMap.put(uid, object);
+		}
+		return objects;
+	}
+
+	protected void setLinks(Resource dataBody, Object instance, ResourceInformation resourceInformation) {
+		String linksFieldName = resourceInformation.getLinksFieldName();
+		if (dataBody.getLinks() != null && linksFieldName != null) {
+			JsonNode linksNode = dataBody.getLinks();
+			Class<?> linksClass = PropertyUtils.getPropertyClass(resourceInformation.getResourceClass(), linksFieldName);
+			ObjectReader linksMapper = objectMapper.readerFor(linksClass);
+			try {
+				Object links = linksMapper.readValue(linksNode);
+				PropertyUtils.setProperty(instance, linksFieldName, links);
+			} catch (IOException e) {
+				throw new ResponseBodyException("failed to parse links information", e);
+			}
+		}
+	}
+
+	protected void setMeta(Resource dataBody, Object instance, ResourceInformation resourceInformation) {
+		String metaFieldName = resourceInformation.getMetaFieldName();
+		if (dataBody.getMeta() != null && metaFieldName != null) {
+			JsonNode metaNode = dataBody.getMeta();
+
+			Class<?> metaClass = PropertyUtils.getPropertyClass(resourceInformation.getResourceClass(), metaFieldName);
+
+			ObjectReader metaMapper = objectMapper.readerFor(metaClass);
+			try {
+				Object meta = metaMapper.readValue(metaNode);
+				PropertyUtils.setProperty(instance, metaFieldName, meta);
+			} catch (IOException e) {
+				throw new ResponseBodyException("failed to parse links information", e);
+			}
+
+		}
+	}
+
+	@Override
+	public boolean isAcceptable(JsonPath jsonPath, String requestType) {
+		throw new IllegalStateException();
```

Comment:
???? This is just a bug. return true/false, which ever is acceptable by default even if not implemented